logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt

```diff
@@ -350,5 +281,83 @@ class ConnectionRepositoryTest {
         )
+
+        fun withSuccessfulCreateConnectionResponse(userId: NetworkUserId): Arrangement {
+            given(connectionApi)
+                .suspendFunction(connectionApi::createConnection)
+                .whenInvokedWith(eq(userId))
+                .then { NetworkResponse.Success(stubConnectionOne, mapOf(), 200) }
+
+            return this
+        }
+
+        fun withErrorOnCreateConnectionResponse(userId: NetworkUserId): Arrangement {
+            given(connectionApi)
+                .suspendFunction(connectionApi::createConnection)
+                .whenInvokedWith(eq(userId))
+                .then { NetworkResponse.Error(KaliumException.GenericError(RuntimeException("An error the server threw!"))) }
+
+            return this
+        }
+
+        fun withErrorOnPersistingConnectionResponse(userId: NetworkUserId): Arrangement {
+            given(conversationDAO)
+                .suspendFunction(conversationDAO::updateOrInsertOneOnOneMemberWithConnectionStatus)
+                .whenInvokedWith(eq(UserIDEntity(userId.value, userId.domain)), any(), any())
+                .thenThrow(RuntimeException("An error occurred persisting the data"))
+
+            return this
+        }
+
+        fun withSuccessfulUpdateConnectionStatusResponse(userId: NetworkUserId): Arrangement {
+            given(connectionApi)
+                .suspendFunction(connectionApi::updateConnection)
+                .whenInvokedWith(eq(userId), eq(ConnectionStateDTO.ACCEPTED))
+                .then { _, _ -> NetworkResponse.Success(stubConnectionOne, mapOf(), 200) }
+            given(conversationDAO)
+                .suspendFunction(conversationDAO::updateOrInsertOneOnOneMemberWithConnectionStatus)
+                .whenInvokedWith(eq(UserIDEntity(userId.value, userId.domain)), any(), any())
+                .then { _, _, _ -> return@then }
+
+            return this
+        }
+
+        fun withErrorUpdatingConnectionStatusResponse(userId: NetworkUserId): Arrangement {
+            given(connectionApi)
+                .suspendFunction(connectionApi::updateConnection)
+                .whenInvokedWith(eq(userId), any())
+                .then { _, _ -> NetworkResponse.Error(KaliumException.GenericError(RuntimeException("An error the server threw!"))) }
+
+            return this
+        }
+
+        fun withSuccessfulResponse(): Arrangement {
+            given(connectionApi)
+                .suspendFunction(connectionApi::fetchSelfUserConnections)
+                .whenInvokedWith(eq(null))
+                .then {
+                    NetworkResponse.Success(
+                        stubConnectionResponse,
+                        mapOf(),
+                        200
+                    )
+                }
+            given(userDetailsApi)
+                .suspendFunction(userDetailsApi::getUserInfo)
+                .whenInvokedWith(any())
+                .then { NetworkResponse.Success(stubUserProfileDTO, mapOf(), 200) }
```

Comment:
The idea is to provide this `stubUserProfileDTO` as a configurable parameter from each specific test instead of hardcoding it inside the function. Otherwise the reader loses control and knowledge over what's happening under the hood on that test config :)

Reply:
good point, done!